Src/Newtonsoft.Json/JsonWriter.cs

```diff
@@ -1576,28 +1576,25 @@ internal static void WriteValue(JsonWriter writer, PrimitiveTypeCode typeCode, o
 #endif
                 default:
 #if !PORTABLE
-                    if (value is IConvertible)
+                    IConvertible convertible = value as IConvertible;
+                    if (convertible != null)
                     {
                         // the value is a non-standard IConvertible
                         // convert to the underlying value and retry
-                        IConvertible convertable = (IConvertible)value;
 
-                        TypeInformation typeInformation = ConvertUtils.GetTypeInformation(convertable);
+                        TypeInformation typeInformation = ConvertUtils.GetTypeInformation(convertible);
```

Comment:
Might as well change the typo on the variable name while changing how it's initialised.